In [1]:
import pandas as pd
import numpy as np

In [3]:
WSV_timeseries_data = pd.read_csv('../data/Evaluating_Happiness/WVS_timeseries_data.csv', low_memory=False)

In [4]:
# see how many rows exist for filtered down wave 6-only data
wave_6_only = WSV_timeseries_data[WSV_timeseries_data['S002'] == 6]

In [5]:
wave_6_only.shape

(89565, 1046)

In [6]:
# how many years covered?
wave_6_only.S020.agg(['min', 'max'])

min    2010
max    2016
Name: S020, dtype: int64

Notes: I'm not sure what's up with this. Suppoedly data collection for this wave took place between 2010 and 2014

In [7]:
# how many in each year?
wave_6_only.S020.value_counts()

2012    28896
2011    25012
2013    17437
2014    10522
2010     5702
2016     1996
Name: S020, dtype: int64

According to https://www.worldvaluessurvey.org/WVSDocumentationWV6.jsp (accessed on 5/25/2021), the 2016 data are all respondents from Haiti. Although dated 2016, their responses are part of the 6th wave of data collection. Double-check this.

In [10]:
wave_6_only[wave_6_only['S020'] == 2016]['S024'].value_counts()

3326    1996
Name: S024, dtype: int64

In [14]:
wave_6_only[wave_6_only['S020'] == 2016]['S003'].value_counts()

332    1996
Name: S003, dtype: int64

Based on the codebook values (the excel version, not the doc), 3326 and 332 are codes for Haiti.